### ISSUE TITLE
Alexander creates “Hello, world” program in Python

### ISSUE THREAD

**Issue by billhiggins** [ayhiggins/experiments#1]
- [x] Set up the Python programming environment
- [x] Write the program
- [x] Commit the program to GitHub

**Comment by billhiggins** [comment]
Steps required for Alexander to set up the Python programming environment

- [x] Create accounts for GitHub and Slack
- [x] Create SSH key for GitHub account
- [x] Install modern Python programming environment
- [x] Install [GitHub Desktop](https://desktop.github.com)
- [x] Install VS Code and Python extension

**Comment by billhiggins** [comment]
About installing a modern Python programming environment …

By default the Mac comes with a back-leveled version of Python—2.6 or something. We want a more modern version of Python, so we’ll install it using a technology called Homebrew, which is a package manager for the Mac.

- [x] Go to [the Homebrew home page](https://brew.sh)
- [x] From the Mac Terminal, follow the instructions to install Homebrew on your Mac
- [x] Install the `python3` package using Homebrew
- [x] Verify that you can start the Python interpreter

**Comment by billhiggins** [comment]
Steps for recreating your primary SSH keypair for github.com:

- [x] Delete the old one from [GitHub](https://github.com/settings/keys)
- [x] Delete the old ones from your computer (under `$HOME/.ssh/id_rsa*`)
- [x] Follow [these instructions](https://help.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) to set it up on your Mac

**Comment by ayhiggins** [comment]
Committed program on June 7, 2020 at 5:50pm

**Comment by billhiggins** [comment]
Adding one final comment to discuss what concepts we explored in this exercise.

**Using WebEx and AirPlay to easily see each other’s work**

You connected to my WebEx and I shared my Mac desktop on the home theater screen so that we could easily see what the other was doing. This makes learning much easier as you can convey information in a much richer way than speaking or Slacking.

**Setting up a working Python development environment on the Mac**

We saw that it takes a great deal of work and care to get to a working development environment on the Mac. Recall that this is because a Mac’s standard configuration is optimized for the needs of an average worker, student, or home user, vs. those of a developer.

- We had to create a few new accounts, for [Slack](https://slack.com/) and for [GitHub](https://github.com/).
- We had to install a few new applications on your Mac, including [Slack](https://slack.com/downloads/), [VSCode](https://code.visualstudio.com/download), and [GitHub Desktop](https://desktop.github.com).
- We had to install the [Homebrew](https://brew.sh) package manager, and then used Homebrew to install a modern version of the [Python](https://www.python.org) programming language and supporting tools.
- We had to [create a new SSH keypair](https://help.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) when we couldn’t recall the passphrase for the keypair we created a few weeks ago.
- Finally, I spent a great deal of time trying to get VSCode configured to use the Homebrew-installed Python environment vs. other older versions floating around your Mac.

**Documenting the work in a GitHub issue**

We rigorously documented our work in [this GitHub issue](https://github.com/ayhiggins/experiments/issues/1), so that in the future we could help reconstruct the tasks and mental models we had at the time of doing this work.

If we worked in a larger team, this would help other team members not directly involved in this work better understand what we did. They would also look at the [changed code](https://github.com/ayhiggins/experiments/commit/ab05d4302ba1c0a10beaf7ea8d67c884d74332b5) for this purpose.

**The Python programming itself was pretty simple**

Once we got to the point of a working development environment, the program itself was quite simple, just [one line of code](https://github.com/ayhiggins/experiments/commit/ab05d4302ba1c0a10beaf7ea8d67c884d74332b5#diff-ae7733612823376ffb6df5a2b7445760).

You ran the command by opening a Terminal within VSCode and typing the command: `python hello_world.py`

Truth be told, we could have successfully written and run this simple program without all of the trouble of setting up a robust development environment, so if anything, this exercise was more about understanding the development environment vs. understanding programming in general or Python in particular.

**Creating the .gitignore file**

Before I let you push your code to github.com, I asked you to create one additional file, `.gitignore`, so that your local VSCode configuration (in the file `.vscode/settings.json`) wouldn’t be shared with “the team” (currently just me), since it’s specific to your computer’s configuration.

**Creating and pushing the commit, to share your code with me**

Finally, when you had successfully written and run your first program, you needed to share it with me by pushing it to github.com. You did this with the following commands:

```
# Mark new files as ready to be committed (this is called “staging” in git)
git add hello_world.py .gitignore

# Package up the staged files into a single, atomic, transferrable commit
git commit -m "Created Hello world program (#1)"

# Upload (“push”) the commit (including the two changed files) to github.com
# so that other team members can download them (“pull”)
git push
```

Note that the first two steps were local to your computer and would have worked even if you’d lacked an Internet connection. The third step used SSH as an application protocol and TCP/IP and various networking infrastructure to transmit your commit as a set of encrypted packets up to github.com. We included the issue number `#1` so that GitHub would create an explicit association between this issue and that commit. You can see how this issue links to the commit two comments above, and the commit itself links to this issue in its subject line.

You can see your first commit containing your first program [here](https://github.com/ayhiggins/experiments/commit/ab05d4302ba1c0a10beaf7ea8d67c884d74332b5). It will be there forever unless you explicitly delete it.

Good job buddy! 🙌🏻
